diff --git a/dynamicserialize/ThriftSerializationContext.py b/dynamicserialize/ThriftSerializationContext.py
index bbfb153..3691b7f 100644
--- a/dynamicserialize/ThriftSerializationContext.py
+++ b/dynamicserialize/ThriftSerializationContext.py
@@ -59,7 +59,7 @@ buildObjMap(dstypes)
 pythonToThriftMap = {
     bytes: TType.STRING,
     int: TType.I32,
-    int: TType.I64,
+    long: TType.I64,
     list: TType.LIST,
     dict: TType.MAP,
     type(set([])): TType.SET,
